### PR TITLE
Re-add support for Chromecast

### PR DESCRIPTION
**Pull request by bsclifton**
Fixes brave/brave-browser#209

***Dependencies***
- [x] https://github.com/brave/brave-core/pull/2493
- [x] https://github.com/brave/devops/pull/1085
- [x] https://github.com/brave/redirector/pull/9
- [x] https://github.com/brave/devops/pull/1293

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [ ] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests && npm run test-security`) on
  - [ ] Windows
  - [ ] macOS
  - [ ] Linux
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [x] Requested a security/privacy review as needed.
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection

## Test Plan:
1. Have a Chrome cast device plugged in and powered on
2. Launch browser
3. Click cast... menu from app menu
4. Check available device list is visible in dialog

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions.

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: lib/whitelistedUrlPrefixes.js [hunk 1]
```diff
@@ -27,5 +27,6 @@ module.exports = [
   'https://ads-serve.bravesoftware.com/', // remove this once ads catalog moves to using prod
   'https://pdfjs.robwu.nl/logpdfjs', // allowed because it gets canceled in tracking protection
   'https://publishers-distro.basicattentiontoken.org/',
-  'https://publishers-staging-distro.basicattentiontoken.org/'
+  'https://publishers-staging-distro.basicattentiontoken.org/',
+  'https://clients2.google.com' // allowed because it 307's to https://extensionupdater.brave.com
```
Comment (jumde) [inline lib/whitelistedUrlPrefixes.js]: extensionupdater.brave.com also needs to be added here.

Comment (bsclifton) [reply lib/whitelistedUrlPrefixes.js]: done!

File: lib/whitelistedUrlPrefixes.js [hunk 2]
```diff
@@ -27,5 +27,7 @@ module.exports = [
   'https://ads-serve.bravesoftware.com/', // remove this once ads catalog moves to using prod
   'https://pdfjs.robwu.nl/logpdfjs', // allowed because it gets canceled in tracking protection
   'https://publishers-distro.basicattentiontoken.org/',
-  'https://publishers-staging-distro.basicattentiontoken.org/'
+  'https://publishers-staging-distro.basicattentiontoken.org/',
+  'https://extensionupdater.brave.com',
```
Comment (diracdeltas) [inline lib/whitelistedUrlPrefixes.js]: please add slash to the end

File: lib/whitelistedUrlPrefixes.js [hunk 3]
```diff
@@ -27,5 +27,7 @@ module.exports = [
...
+  'https://clients2.google.com' // allowed because it 307's to https://extensionupdater.brave.com
```
Comment (diracdeltas) [inline lib/whitelistedUrlPrefixes.js]: same here

File: lib/whitelistedUrlPrefixes.js (hunk 2 above)
Comment (bsclifton) [reply lib/whitelistedUrlPrefixes.js]: Fixed!

File: lib/whitelistedUrlPrefixes.js (hunk 3 above)
Comment (bsclifton) [reply lib/whitelistedUrlPrefixes.js]: Fixed!

File: lib/whitelistedUrlPrefixes.js [hunk 4]
```diff
@@ -27,5 +27,7 @@ module.exports = [
   'https://ads-serve.bravesoftware.com/', // remove this once ads catalog moves to using prod
   'https://pdfjs.robwu.nl/logpdfjs', // allowed because it gets canceled in tracking protection
   'https://publishers-distro.basicattentiontoken.org/',
-  'https://publishers-staging-distro.basicattentiontoken.org/'
+  'https://publishers-staging-distro.basicattentiontoken.org/',
+  'https://extensionupdater.brave.com/',
+  'https://clients2.google.com/' // allowed because it 307's to https://extensionupdater.brave.com
```
Comment (diracdeltas) [inline lib/whitelistedUrlPrefixes.js]: can we be more specific with this prefix based on https://github.com/brave/brave-core/pull/2493#issuecomment-495721065?

for instance `https://clients2.google.com/service/update2/crx`

**Comment by mihaiplesa** [comment]
Has failed network audit, CI log at https://staging.ci.brave.com/view/ci/job/brave-browser-build-pr/view/change-requests/job/PR-4631/lastSuccessfulBuild/execution/node/413/log/

**Comment by bsclifton** [comment]
CLOSING - this is no longer needed 😄

**Pull request closed without merging**